src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java

```diff
@@ -125,12 +125,13 @@ private void createLatestSymlink(Plugin hpi) throws IOException {
         if (System.getProperty("os.name").toLowerCase(Locale.US).contains("windows")) {
             return;
         }
-        pb.command("ln", "-s", hpi.getLatest().version, "latest");
+        final String version = hpi.getLatest().version;
+        pb.command("ln", "-s", version, "latest");
         pb.directory(dir);
         try {
             int r = pb.start().waitFor();
             if (r != 0) {
-                throw new IOException("ln failed: " + r); // TODO better logging
+                throw new IOException("ln failed in: '" + dir + "' to: '" + version + "' exit code: " + r); // TODO better logging
```

Comment:
```suggestion
                throw new IOException("ln failed in: '" + dir + "' to: '" + version + "' exit code: " + r + "\nError: " + IOUtils.toString(pb.getErrorStream()) + "\nOutput: " + IOUtils.toString(pb.getInputStream());
```
From https://github.com/jenkins-infra/update-center2/blob/278420dfa9564fbe00cd768a972d48d760d50430/src/main/java/io/jenkins/update_center/DirectoryTreeBuilder.java#L168-L169